### ISSUE TITLE
History compare/diff is broken

### ISSUE THREAD

**Issue by mck9** [plone/volto#982]
Volto 3.2.7
Plone 5.1.6

When trying to view the diff of a former revision, volto throws an error:

Sorry, something went wrong with your request 
value is null

```
    in DiffField (at Diff.jsx:298)
    in div (created by Container)
    in Container (at Diff.jsx:217)
    in DiffComponent (created by ConnectFunction)
    in ConnectFunction (created by InjectIntl(Connect(DiffComponent)))
    in InjectIntl(Connect(DiffComponent)) (created by Context.Consumer)
    in Route (created by withRouter(InjectIntl(Connect(DiffComponent))))
    in withRouter(InjectIntl(Connect(DiffComponent))) (created by Context.Consumer)
    in Route (created by AppComponent)
    in Switch (created by AppComponent)
    in main (at App.jsx:121)
    in div (created by Segment)
    in Segment (at App.jsx:120)
    in AppComponent (created by ConnectFunction)
    in ConnectFunction (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in Route (created by Context.Consumer)
    in Switch (created by Context.Consumer)
    in Route (created by AsyncConnect)
    in AsyncConnect (created by Context.Consumer)
    in AsyncConnectWithContext (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in Route (created by withRouter(Connect(AsyncConnectWithContext)))
    in withRouter(Connect(AsyncConnectWithContext)) (at start-client.jsx:32)
    in ScrollToTop (created by Context.Consumer)
    in Route (created by withRouter(ScrollToTop))
    in withRouter(ScrollToTop) (at start-client.jsx:31)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by ConnectFunction)
    in ConnectFunction (at start-client.jsx:30)
    in IntlProvider (created by ConnectFunction)
    in ConnectFunction (created by Provider)
    in Provider (created by Provider)
    in Provider (at start-client.jsx:29)
```

**Comment by mck9** [comment]
On volto 4.0.0-alpha.10 / Plone 5.2 volto sometimes shows the diff page, but it is empty or some random error is thrown.

The Plone server logs this error:

`aborting transaction due to no CSRF protection on url http://localhost:8080/Plone/test-block/@history/3`

**Comment by robdayz** [comment]
I was checking this error. And it appears to me too, I have a related problem because the split function inside react-dom is splitting more than needed, and I can't read the error log :(  Seems that the history view from Plone takes the Text field  (https://github.com/plone/plone.app.contenttypes/blob/127b18e5ee8685f421f3adb3c5657ac2b65a1585/plone/app/contenttypes/browser/templates/document.pt#L14) to make the history diff, but when it is empty then throws the error. I will continue on that.

**Comment by avoinea** [comment]
Step to reproduce (**Volto 4.0.0-alpha.42** & **Plone 5.2.1**):
1. Add & Save Page with title
2. Edit & Add Block
3. See History & Click Edited

https://volto.kitconcept.com/page/diff?one=0&two=1


```
TypeError: Cannot read property 'pathname' of undefined
(anonymous function)
node_modules/@plone/volto/src/components/theme/View/DefaultView.jsx:54
  51 |     id={block}
  52 |     properties={content}
  53 |     data={content[blocksFieldname][block]}
> 54 |     path={getBaseUrl(location.pathname)}
     | ^  55 |   />
  56 | ) : (
  57 |   <div key={block}>
```

**Comment by avoinea** [comment]
I'll take a look into this...

**Comment by avoinea** [comment]
Fixed and merged by #1242
